File: libraries/typescript/src/tts/gemini.ts
/** Gemini TTS for Patter pipeline mode. */
import { GeminiTTS as _GeminiTTS } from "../providers/gemini-tts";

/** Constructor options for the Gemini `TTS` adapter. */
export interface GeminiTTSOptions {
  /** API key. Falls back to GEMINI_API_KEY / GOOGLE_API_KEY env var when omitted. */
  apiKey?: string;
  /** Prebuilt voice name (default `Kore`). */
  voice?: string;
  /** TTS model (default `gemini-3.1-flash-tts-preview`). */
  model?: string;
  /** Output PCM rate: 8000 or 16000 (default 16000). */
  targetSampleRate?: number;
}

/**
 * Gemini TTS (`gemini-3.1-flash-tts-preview`).
 *
 * @example
 * ```ts
 * import * as gemini from "getpatter/tts/gemini";
 * const tts = new gemini.TTS();                       // reads GEMINI_API_KEY
 * const tts = new gemini.TTS({ voice: "Kore" });
 * ```
 */
export class TTS extends _GeminiTTS {
  static readonly providerKey = "gemini_tts";
  constructor(opts: GeminiTTSOptions = {}) {
    const key = opts.apiKey ?? process.env.GEMINI_API_KEY ?? process.env.GOOGLE_API_KEY;
    if (!key) {
      throw new Error(
        "Gemini TTS requires an apiKey. Pass { apiKey: '...' } or set " +
          "GEMINI_API_KEY / GOOGLE_API_KEY in the environment.",
      );
    }
    super(key, opts.voice ?? "Kore", opts.model ?? "gemini-3.1-flash-tts-preview", opts.targetSampleRate ?? 16000);
  }
}

File: libraries/python/getpatter/tts/gemini.py
"""Gemini TTS for Patter pipeline mode (Beta)."""

from __future__ import annotations

from typing import ClassVar

from getpatter.providers.gemini_tts import (
    GEMINI_TTS_DEFAULT_MODEL,
    GEMINI_TTS_DEFAULT_TARGET_SAMPLE_RATE,
    GEMINI_TTS_DEFAULT_VOICE,
    GeminiTTS as _GeminiTTS,
)

__all__ = ["TTS"]


class TTS(_GeminiTTS):
    """Gemini TTS (``gemini-3.1-flash-tts-preview``) (Beta).

    Example::

        from getpatter.tts import gemini

        tts = gemini.TTS()                        # reads GEMINI_API_KEY
        tts = gemini.TTS(api_key="...", voice="Kore")

    Inline delivery tags in the text — ``[warm]``, ``[short pause]``,
    ``[sigh]`` — shape prosody instead of being spoken.
    """

    provider_key: ClassVar[str] = "gemini_tts"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        voice: str = GEMINI_TTS_DEFAULT_VOICE,
        model: str = GEMINI_TTS_DEFAULT_MODEL,
        target_sample_rate: int = GEMINI_TTS_DEFAULT_TARGET_SAMPLE_RATE,
    ) -> None:
        super().__init__(
            api_key=api_key,
            voice=voice,
            model=model,
            target_sample_rate=target_sample_rate,
        )
